src/neo-vm/ExecutionEngine.cs

```diff
@@ -277,17 +277,23 @@ private bool ExecuteInstruction()
                         if (!ExecuteCall(x.Position)) return false;
                         break;
                     }
-                case OpCode.THROW:
+                case OpCode.ABORT:
                     {
                         return false;
                     }
-                case OpCode.THROWIF:
-                case OpCode.THROWIFNOT:
+                case OpCode.ASSERT:
                     {
                         if (!TryPop(out bool x)) return false;
-                        if (x ^ (instruction.OpCode == OpCode.THROWIFNOT)) return false;
+                        if (!x)
+                        {
+                            return false;
+                        }
                         break;
                     }
+                case OpCode.THROW:
```

Comment:
What it's the difference between `THROW` and `ABORT`?

Reply:
`ABORT` will set vm state to FAULT immediately, and cannot be caught.

Reply:
Then... we are mixing pull requests, `try/catch` and `ASSERT`. But it's ok